service/resource/legacy/resource.go

```diff
@@ -85,35 +90,73 @@ func New(config Config) (*Resource, error) {
 }
 
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for the daemon set in the Kubernetes API")
+
+	var currentDaemonSet *v1beta1.DaemonSet
+	{
+		manifest, err := r.k8sClient.Extensions().DaemonSets(networkNamespace(customObject.Spec)).Get(networkApp, apismetav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			r.logger.Log("cluster", key.ClusterID(customObject), "debug", "did not find the daemon set in the Kubernetes API")
+			// fall through
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		} else {
+			r.logger.Log("cluster", key.ClusterID(customObject), "debug", "found the daemon set in the Kubernetes API")
+			currentDaemonSet = manifest
+			r.updateVersionBundleVersionGauge(customObject, versionBundleVersionGauge, currentDaemonSet)
+		}
+	}
+
+	return currentDaemonSet, nil
```

Comment:
This is a proper current state implementation of the daemon set. So a good step into the right direction. Note that right now the current state returned here is not used for anything. For now we just get the metrics magic done and this is good enough for me now.